### PR TITLE
Use TransformingClassLoader in Tests and Write Data-Driven Enum Extension Tests

### PR DESCRIPTION
**Pull request by shartte**
Change config key from `enumExtender` to `enumExtensions` since this is still the chance to have it named according to what the setting does.
The JSON file only contains extensions to enums in a declarative manner, which means `enumExtensions` is a better name.

### CODE REVIEW

File: loader/src/main/java/net/neoforged/fml/loading/LoadingModList.java [hunk 1]
```diff
@@ -92,9 +92,9 @@ public void addEnumExtenders() {
         modFiles.stream()
                 .map(ModFileInfo::getMods)
                 .flatMap(List::stream)
-                .forEach(mod -> mod.getConfig().<String>getConfigElement("enumExtender").ifPresent(file -> {
+                .forEach(mod -> mod.getConfig().<String>getConfigElement("enumExtensions").ifPresent(file -> {
```
Comment (marchermans) [inline loader/src/main/java/net/neoforged/fml/loading/LoadingModList.java:95]: Why was this changed?

File: loader/src/main/java/net/neoforged/fml/common/asm/enumextension/RuntimeEnumExtender.java [hunk 2]
```diff
@@ -452,7 +452,7 @@ public static void loadEnumPrototypes(Map<IModInfo, Path> paths) {
                 .reduce(
                         new HashMap<>(),
                         (map, proto) -> {
-                            map.computeIfAbsent(proto.enumName(), $ -> new ArrayList<>()).add(proto);
```
Comment (Matyrobbrt) [inline loader/src/main/java/net/neoforged/fml/common/asm/enumextension/RuntimeEnumExtender.java:455]: You really hate money :P

File: loader/src/main/java/net/neoforged/fml/loading/LoadingModList.java [hunk 3]
```diff
@@ -92,9 +92,9 @@ public void addEnumExtenders() {
...
                     Path path = mod.getOwningFile().getFile().findResource(file);
-                    if (Files.notExists(path)) {
+                    if (!Files.isRegularFile(path)) {
                         ModLoader.addLoadingIssue(ModLoadingIssue.error("fml.modloading.enumextender.file_not_found", path).withAffectedMod(mod));
```
Comment (Matyrobbrt) [inline loader/src/main/java/net/neoforged/fml/loading/LoadingModList.java:98]: I feel like it should have a better error message if for some reason it's a directory.

Comment (Matyrobbrt) [reply loader/src/main/java/net/neoforged/fml/loading/LoadingModList.java:98]: Very well then

**Review by Matyrobbrt — APPROVED** [review]
